### PR TITLE
patterns

### PR DESCRIPTION
**Pull request by rkrylov**

### CODE REVIEW

**Comment by TrueJugernaut** [comment]
Lgfm

File: src/main/java/Patterns/FactoryMethod/BuilderPattern.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public String toString() {
+        return "UserClass{" +
+                "name='" + name + '\'' +
+                '}';
+    }
+}
+
+class UserClassBuilder {
```
Comment (boroda4436) [inline src/main/java/Patterns/FactoryMethod/BuilderPattern.java]: add example of using UserClassBuilder

Comment (boroda4436) [reply src/main/java/Patterns/FactoryMethod/BuilderPattern.java]: add two more fields to UserClass

File: src/main/java/Patterns/FactoryMethod/SingletonPattern.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package Patterns.FactoryMethod;
+
+public class SingletonPattern implements Pattern {
```
Comment (boroda4436) [inline src/main/java/Patterns/FactoryMethod/SingletonPattern.java:3]: new SingletonPattern() should be private

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/Patterns/Main.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        System.out.println("\n");
+        new Flyweight().testFlyweight();
+
+        System.out.println("\n");
+        new Prototype().prototypeTest();
+
+        System.out.println("\n");
+        new AbstractFactory().testAbstractFabric();
+
+        UserClass userClass = new UserClass.Builder()
```
Comment (boroda4436) [inline src/main/java/Patterns/Main.java:38]: +1

**Review by boroda4436 — APPROVED** [review]
LGTM

**Pull request merged**